tools/class-parse/class-parse.csproj

```diff
@@ -1,7 +1,7 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <TargetFrameworks>net472;net6.0</TargetFrameworks>
+    <TargetFrameworks>net472;$(DotNetStableTargetFramework)</TargetFrameworks>
```

Comment:
I made a thing like `$(DotNetStableTargetFramework)` in xamarin-android for console apps that are used during our build. Should utilities we ship like `class-parse` actually build against 7? It is invoked from MSBuild in customer projects, but they should have a .NET 7 SDK.

We can also wait and update these later.

Reply:
I figured `$(DotNetStableTargetFramework)` would also be useful in JI for certain projects, but using it more widely was a bit of a hassle. Keeping the tools that XA ships on a stable .NET version gives us the flexibility to not require a system wide .NET 7 install in XA for now. If we bump these to net7.0 XA will require a global net7.0 install, or we'll need to update our build tasks to run the tool with our local dotnet install. I don't mind bumping these to net7.0 now but I didn't see a strong need to do so, but maybe I am not considering something.

Reply:
I'm fine with whatever is easiest for now, but we should endeavor to get everything we ship running on .NET 7 previews since that is how our customers will eventually receive it.  We need to find any issues caused by .NET 7 before they do.  😁

Reply:
xamarin-android/main is already running these on .NET 7, they just have a `TargetFramework` of `net6.0`.

If you look at a `.binlog` of one of our MSBuild tests, it's using a local .NET 7 SDK we provision at `./bin/Release/dotnet/dotnet`.

These definitely work on 7, because 6 isn't in here at all:

![image](https://user-images.githubusercontent.com/840039/170407568-b8145aa5-ccda-425d-b322-64d9cf7fe262.png)

![image](https://user-images.githubusercontent.com/840039/170407590-268f3bd6-51a5-40f9-a52f-d3535e0d2d92.png)

Reply:
I didn't look too closely, but I think the [AndroidDotnetToolTask](https://github.com/xamarin/xamarin-android/blob/de5c8f89d18fcc8b39dd8e6ec1ccf51724dbadd0/src/Xamarin.Android.Build.Tasks/Tasks/AndroidDotnetToolTask.cs#L73) that runs tools like generator and class parse only uses the `dotnet` that is found in path? In our XA build case where we test out of a "sandboxed" location I think we end running these tools with the latest version installed system wide. We may want to update that task to try to detect and use the `dotnet` path that the project is being built from, assuming this is the case. I can investigate this further outside of this PR though.

Reply:
I think there must be some logic when you `Process.Start("dotnet")` from inside a `dotnet` process, for example `dotnet run` just does this:

https://github.com/dotnet/sdk/blob/ad1ca90ad286288970a4ae1ec77e85848fef5a24/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.targets#L869-L877